Linking error in DataFormats/Headers/

When building a fresh copy of O2 using alibuild, i get the following linking error:

```
DEBUG:O2:0: CMakeFiles/fairMQmonitor.dir/src/FairMQmonitor.cxx.o: In function `AliceO2::Header::Block::freefn(void*, void*)':
DEBUG:O2:0: /opt/alice/sw/SOURCES/O2/dev/0/DataFormats/Headers/include/Headers/DataHeader.h:301: undefined reference to `AliceO2::Header::Block::sDeleter'
DEBUG:O2:0: collect2: error: ld returned 1 exit status
```

Ubuntu 14.04.5
GCC 4.8.4
Alidist branch IB/v5-08/o2

@mkrzewic

--- a/DataFormats/Headers/include/Headers/DataHeader.h
+++ b/DataFormats/Headers/include/Headers/DataHeader.h
@@ -405,7 +405,7 @@ const AliceO2::Header::HeaderType NameHeader<N>::sHeaderType = "NameHead";
 // dirty trick to always have access to the headertypeID of a templated header type
 // TODO: find out if this can be done in a nicer way + is this realy necessary?
 template <>
-const AliceO2::Header::HeaderType NameHeader<0>::sHeaderType = "NameHead";
+const AliceO2::Header::HeaderType NameHeader<0>::sHeaderType;
 
 template <size_t N>
 const SerializationMethod NameHeader<N>::sSerializationMethod = gSerializationMethodNone;

--- a/DataFormats/Headers/src/DataHeader.cxx
+++ b/DataFormats/Headers/src/DataHeader.cxx
@@ -54,6 +54,10 @@ const uint32_t AliceO2::Header::DataHeader::sVersion = 1;
 const AliceO2::Header::HeaderType AliceO2::Header::DataHeader::sHeaderType = String2uint64("DataHead");
 const AliceO2::Header::SerializationMethod AliceO2::Header::DataHeader::sSerializationMethod = AliceO2::Header::gSerializationMethodNone;
 
+//storage fr NameHeader static
+template <>
+const AliceO2::Header::HeaderType AliceO2::Header::NameHeader<0>::sHeaderType = "NameHead";
+
 using namespace AliceO2::Header;
 
 //__________________________________________________________________________________________________

--- a/DataFormats/Headers/src/DataHeader.cxx
+++ b/DataFormats/Headers/src/DataHeader.cxx
@@ -41,6 +41,9 @@ const AliceO2::Header::DataDescription AliceO2::Header::gDataDescriptionTracks  
 const AliceO2::Header::DataDescription AliceO2::Header::gDataDescriptionConfig  ("CONFIG         ");
 const AliceO2::Header::DataDescription AliceO2::Header::gDataDescriptionInfo    ("INFO           ");
 
+//definitions for Block statics
+std::default_delete<byte[]> AliceO2::Header::Block::sDeleter;
+
 //storage for BaseHeader static members, all invalid
 const uint32_t AliceO2::Header::BaseHeader::sVersion = AliceO2::Header::gInvalidToken32;
 const AliceO2::Header::HeaderType AliceO2::Header::BaseHeader::sHeaderType = AliceO2::Header::gInvalidToken64;

--- a/Utilities/O2device/include/O2device/O2device.h
+++ b/Utilities/O2device/include/O2device/O2device.h
@@ -47,11 +47,10 @@ public:
     AliceO2::Header::Block headerBlock{std::move(incomingBlock)};
     FairMQMessagePtr dataMessage{std::move(incomingDataMessage)};
 
-    FairMQMessagePtr headerMessage =
-      fTransportFactory->CreateMessage(headerBlock.buffer.get(),
-                                       headerBlock.bufferSize,
-                                       &AliceO2::Header::Block::freefn,
-                                       headerBlock.buffer.get());
+    FairMQMessagePtr headerMessage = NewMessage(headerBlock.buffer.get(),
+                                                headerBlock.bufferSize,
+                                                &AliceO2::Header::Block::freefn,
+                                                headerBlock.buffer.get());
     headerBlock.buffer.release();
 
     parts.AddPart(std::move(headerMessage));

--- a/Utilities/O2device/include/O2device/O2device.h
+++ b/Utilities/O2device/include/O2device/O2device.h
@@ -47,10 +47,12 @@ public:
     AliceO2::Header::Block headerBlock{std::move(incomingBlock)};
     FairMQMessagePtr dataMessage{std::move(incomingDataMessage)};
 
-    auto buffer = headerBlock.buffer.release();
-    auto size = headerBlock.bufferSize;
     FairMQMessagePtr headerMessage =
-      fTransportFactory->CreateMessage(buffer, size, &AliceO2::Header::Block::freefn, buffer);
+      fTransportFactory->CreateMessage(headerBlock.buffer.get(),
+                                       headerBlock.bufferSize,
+                                       &AliceO2::Header::Block::freefn,
+                                       headerBlock.buffer.get());
+    headerBlock.buffer.release();
 
     parts.AddPart(std::move(headerMessage));
     parts.AddPart(std::move(dataMessage));

